doc/sphinx/practical-tools/utilities.rst

```diff
@@ -1,53 +1,371 @@
+  projects need another mechanism.
+
+.. _COQPATH:
+
+- The `COQPATH` environment variable, which contains a list of directories.
+  Each of their subdirectories is a separately-compiled package.  The entire
+  logical path is derived from the directory name (as for `Bignums` above).
+  See :ref:`customization-by-environment-variables`.
+
+  .. todo improve what's written at the hyperlink
+
+- :cmd:`Add LoadPath` adds a new logical path/physical path pair.  We recommend
+  you avoid using :cmd:`Add LoadPath` because embedding physical paths in your scripts
+  makes it harder to install and use them on another computer.
+
+We recommend using `COQPATH`.  For simpler developments, it could have a single
+entry pointing to a directory that contains your various separate projects,
+each in its own subdirectory.  For more advanced
+development work, you may want to set `COQPATH` differently for different projects
+or different versions of a project.  Currently, if you want to change `COQPATH`
+while you're using CoqIDE, you must restart CoqIDE or start another CoqIDE process
+after making the change.  Perhaps in the future CoqIDE will show the current
+setting for and allow changing `COQPATH`.
+
+  .. note ::
+     Packages that have only the directory name ``.`` in ``-R <dirname> <dirpath>``
+     have logical names that map exactly to the directory names.  Including such
+     packages in a directory in `COQPATH` will work fine.
+
+     In contrast, packages that have a directory name other than `.` in `-R`
+     have subdirectories that don't become part of the :term:`logical path`.  If they're
+     installed in the Coq directories, they're fine.  If they're in `COQPATH`,
+     the `_CoqProject` file is essential to getting the correct logical paths within the packages.
+     Since Coq tools currently support only  a single `_CoqProject` file, such packages
+     work correctly only if the IDE is running a script that's part of that package.
+
+     Perhaps Coq will eventually support multiple `_CoqProject` files at once to eliminate
+     this limitation.
+
+Alternatively, if you have a single-package project that depends on external
+packages that were not installed with the Coq binaries or accessible through
+`COQPATH`, you could add `-Q` options to the `_CoqProject` that point to the
+dependencies.  Of course, this makes your `_CoqProject` system dependent.
+Those added options shouldn't appear in published packages.
+
+.. todo I thought @herbelin added code to complain about ambiguous short names
+   I made up some stuff below, need to check it:
+
+If you associate the same logical name with more than one directory, Coq will
+use the last one added (i.e., that appears earlier in :cmd:`Print LoadPath`).
+Nor will Coq complain if the short name of a package is ambiguous.  Perhaps
+in the future, Coq will be able to report such conflicts.  We recommend using
+the :n:`From @dirpath` option of :cmd:`Require` to make the behavior more predictable.
```

Comment:
It reports an ambiguity for `Require` but not for `From ... Require`.  See #16314.  When I do the equivalent with `_CoqProject`, there's no visible error, either:

```
$ more _CoqProject
-R . Ambig
a.v
ambig2/a.v
$ make -f CoqMakefile
COQC a.v
COQC ambig2/a.v
$ coqc --version
The Coq Proof Assistant, version 8.15.2
```

I will reword a bit.

Reply:
As I explained in the issue, that is not really ambiguous. What would be ambiguous for both `From ... Require` and `Require` would be:
```
A/B/a.v
A/C/a.v
```

and you wrote one of:
```
Require a.
From A Require a.
```

Reply:
> The actual specification of what is considered ambiguous however is very complicated and not fully written out.

Not written out at all.  At the very least, the `Require` command should describe the behavior so users understand it.  The `Require` doc doesn't mention "shortest" anywhere so the behavior contradicts the description.  The complexity may be justifiable for compatibility reasons and to avoid the need for pervasive changes to `Require` commands.  If someone can describe the behavior soon I'll include that in this PR.

Reply:
@jfehrle I don't think it is a simple task describing the behaviour of the `From ... Require` command. It has a very subtle semantics. We can paint a picture of some of the important parts with some broad strokes, but a full description of how the algorithm works will be difficult to compile.

Historically, the technical difficulty and subtlety of this algorithm has been the main barrier to its documentation. Hugo tried to at least coincide the behaviours of coqdep and coqc, together with writing many tests demonstrating the expected behaviour. I believe it would be an entire project onto itself to describe with further details what is going on. Not something I am against starting however.

Reply:
I removed lines 342-344.  I understand it's an exact match, which would be a shortest match, but "shortest" doesn't apply in general.